docs/auth/router-guards.md

```diff
@@ -0,0 +1,58 @@
+# Route users with AngularFire guards
+
```

Comment:
For a "what this solves" blurb at the top, maybe something like "These are convenience pipes provided by AngularFire to protect routes using Angular's [Route Guards](https://angular.io/guide/router#milestone-5-route-guards)."